diff --git a/hw/xfree86/os-support/linux/systemd-logind.c b/hw/xfree86/os-support/linux/systemd-logind.c
index 13784d1..661fde5 100644
--- a/hw/xfree86/os-support/linux/systemd-logind.c
+++ b/hw/xfree86/os-support/linux/systemd-logind.c
@@ -499,6 +499,7 @@ connect_hook(DBusConnection *connection, void *data)
         goto cleanup;
     }
 
+
     dbus_bus_add_match(connection,
         "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',path='/org/freedesktop/DBus'",
         &error);
@@ -526,6 +527,28 @@ connect_hook(DBusConnection *connection, void *data)
         goto cleanup;
     }
 
+    msg = dbus_message_new_method_call("org.freedesktop.login1",
+            session, "org.freedesktop.login1.Session", "SetType");
+    if (!msg) {
+        LogMessage(X_ERROR, "systemd-logind: out of memory\n");
+        goto cleanup;
+    }
+
+    const char *type;
+    type = "x11";
+    if (!dbus_message_append_args(msg, DBUS_TYPE_STRING, &type,
+                                  DBUS_TYPE_INVALID)) {
+        LogMessage(X_ERROR, "systemd-logind: out of memory\n");
+        goto cleanup;
+    }
+
+    reply = dbus_connection_send_with_reply_and_block(connection, msg,
+                                                      DBUS_TIMEOUT_USE_DEFAULT, &error);
+    if (!reply) {
+        LogMessage(X_ERROR, "systemd-logind: SetType failed: %s\n",
+                   error.message);
+        goto cleanup;
+    }
     /*
      * HdG: This is not useful with systemd <= 208 since the signal only
      * contains invalidated property names there, rather than property, val
